### ISSUE TITLE
How to get block author

### ISSUE THREAD

**Issue by bizzyvinci** [polkadot-js/api#4795]
I'm trying to get block author from moonbeam. And here my failed attempts. Tested version is `7.5.1`

```ts
import { ApiPromise, WsProvider } from "@polkadot/api";
import { extractAuthor } from "@polkadot/api-derive/type/util";
import { SignedBlock } from "@polkadot/types/interfaces";

const wsProvider = new WsProvider(
  "wss://moonbeam.api.onfinality.io/public-ws"
);
const api = await ApiPromise.create({ provider: wsProvider });
```

1. `author` is not present in block `header` using `api.rpc.chain.getBlock` nor `api.derive.chain.getHeader` as shown in the [documentation](https://polkadot.js.org/docs/api/cookbook/blocks#how-do-i-retrieve-the-headerextrinsic-hash-from-blocks)
2.  Using [extractAuthor](https://github.com/polkadot-js/api/blob/master/packages/api-derive/src/type/util.ts#L6) with the snippet below. block and validators would be logged but author is undefined.
```ts
const signedBlock = await api.rpc.chain.getBlock(
  "0xcc8e3407ec9cd950ef9102e080e8be4642a66c2a01d587d09c11d89e866a6214"
);
console.log(signedBlock)
const block = signedBlock as SignedBlock;
console.log(block);

const candidatePool = JSON.parse(
  (await api.query.parachainStaking.candidatePool()).toString()
);
console.log(candidatePool);

const validators = candidatePool.map((x: { owner: any }) => x.owner);
console.log(validators);

const author = extractAuthor(block.block.header.digest, validators);
console.log(author);
```

**Comment by jacogr** [comment]
Moonbeam has special logic, not compatible with the base Substrate - so in this case, cannot comment if it is not outdated *(it has not been adjusted by that team). This _may_ give info, if not, no additional ideas - https://github.com/polkadot-js/api/blob/master/packages/api-derive/src/type/HeaderExtended.ts#L21 

(like mentioned, the digest, eg. are different in this specific implementation, so cannot make any guarantees apart from what is contributes)

TL;DR Use api.derive.chain.*, if not applicable, refer to the logic - which may be outdated by now (as contributed)

**Comment by bizzyvinci** [comment]
moonbeam author can be found in `api.rpc.eth.getBlockByNumber` or `api.rpc.eth.getBlockByHash`. Is there a way to integrate it so it's accessible in other substrate-like ways e.g `api.derive.chain.*`?

**Comment by jacogr** [comment]
For Moonbeam, this is what the team contributed to the apps UI to extract the author - https://github.com/polkadot-js/apps/blob/c065c7b074b61c8a5ac07bba46a84a41ef5d150c/packages/react-query/src/BlockAuthors.tsx#L69-L76

**Comment by bizzyvinci** [comment]
Thanks @jacogr. The code snippet helped me to resolve my issue. `api.query.authorMapping` is allowed in my sandbox and I've been able to make it work.

**Comment by bizzyvinci** [comment]
This issue can be closed... But if there's a need and way to add moonbeam logic in codebase, I'm willing to help with guidance.
